azurelinuxagent/common/rdma.py

```diff
@@ -90,3 +96,133 @@ def reboot_system(self):
+            logger.info("RDMA: trying {0}".format(f))
+            if not os.path.isfile(f):
+                logger.info(
+                    "RDMA: DAPL config not found at {0}".format(f))
+                continue
+            logger.info("RDMA: DAPL config is at: {0}".format(f))
+            cfg = fileutil.read_file(f)
+            new_cfg = RDMADeviceHandler.replace_dat_conf_contents(
+                cfg, ipv4_addr)
+            fileutil.write_file(f, new_cfg)
+            logger.info("RDMA: DAPL configuration is updated")
+            return
+
+        raise Exception("RDMA: DAPL configuration file not found at predefined paths")
+
+    @staticmethod
+    def replace_dat_conf_contents(cfg, ipv4_addr):
+        old = "ofa-v2-ib0 u2.0 nonthreadsafe default libdaplofa.so.2 dapl.2.0 \"\S+ 0\""
+        new = "ofa-v2-ib0 u2.0 nonthreadsafe default libdaplofa.so.2 dapl.2.0 \"{0} 0\"".format(
+            ipv4_addr)
+        return re.sub(old, new, cfg)
+
+    @staticmethod
+    def write_rdma_config_on_device(path, ipv4_addr, mac_addr):
+        data = RDMADeviceHandler.get_rdma_config(ipv4_addr, mac_addr)
+        logger.info(
+            "RDMA: Updating device with configuration: {0}".format(data))
+        with open(path, "w") as c:
+            c.write(data)
+        logger.info("RDMA: Updated device with IPv4/MAC addr successfully")
+
+    @staticmethod
+    def get_rdma_config(ipv4_addr, mac_addr):
+        return 'rdmaMacAddress="{0}" rdmaIPv4Address="{1}"'.format(mac_addr, ipv4_addr)
+
+    @staticmethod
+    def wait_rdma_device(path, timeout_sec, check_interval_sec):
+        logger.info("RDMA: waiting for device {0}".format(path))
+        total_retries = timeout_sec/check_interval_sec
+        n = 0
+        while n < total_retries:
+            if os.path.exists(path):
+                logger.info("RDMA: device ready")
+                return
+            logger.info(
+                "RDMA: device not ready, sleep {0}s".format(check_interval_sec))
+            time.sleep(check_interval_sec)
+            n += 1
+        logger.error("RDMA device wait timed out")
+        raise Exception("The device did not show up in {0} seconds ({1} retries)".format(
+            timeout_sec, total_retries))
+
+    @staticmethod
+    def update_network_interface(mac_addr, ipv4_addr):
+        netmask=16
+        
+        logger.info("RDMA: will update the network interface with IPv4/MAC")
+
+        if_name=RDMADeviceHandler.get_interface_by_mac(mac_addr)
+        logger.info("RDMA: network interface found: {0}", if_name)
+        logger.info("RDMA: bringing network interface up")
+        if shellutil.run("ifconfig {0} up".format(if_name)) != 0:
+            raise Exception("Could not bring up RMDA interface: {0}".format(if_name))
+
+        logger.info("RDMA: configuring IPv4 addr and netmask on interface")
+        addr = '{0}/{1}'.format(ipv4_addr, netmask)
+        if shellutil.run("ifconfig {0} {1}".format(if_name, addr)) != 0:
```

Comment:
I am wondering if we need to worry about older distributions. Presumably the kernel module only works on reasonably new distributions that also support the "ip" commands. Given that "ifconfig" is considered somewhat antiquated it might be nice to swich over to us "ip" instead of "ifconfig"